### PR TITLE
feat(kubernetes) add kubernetes official docs support

### PR DESCRIPTION
**Pull request by rpvsilva**
## Description

This extension allows searching in the Kubernetes official documentation through the raycast.

### Extention Screenshot
Screenshot 2022-02-20 at 14 43 08

### Search List
Screenshot 2022-02-20 at 14 43 33

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)

### CODE REVIEW

File: extensions/kube-docs/package.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
+{
+  "$schema": "https://www.raycast.com/schemas/extension.json",
+  "name": "kubernetes-docs",
```
Comment (pernielsentikaer) [inline extensions/kube-docs/package.json:3]: Could the folder renamed to `kubernetes-docs`?

File: extensions/kube-docs/package.json [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  "title": "Kubernetes Documenation",
+  "description": "Get kubernetes official documentation",
```
Comment (pernielsentikaer) [inline extensions/kube-docs/package.json:5]:
```suggestion
  "description": "Search kubernetes official documentation",
```

File: extensions/kube-docs/package.json [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  "icon": "command-icon.png",
+  "author": "rpvsilva",
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "index",
+      "title": "Search Documentation",
+      "subtitle": "Kubernetes",
+      "description": "Kubernetes Documentation",
```
Comment (pernielsentikaer) [inline extensions/kube-docs/package.json:14]:
```suggestion
      "description": "Search in Kubernetes documentation",
```

File: extensions/kube-docs/src/index.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    const {
+      data: { webPages },
+    } = await fetcher<DocsResponse>({ query });
+
+    if (!webPages || !webPages.value) return;
+
+    setItems(webPages.value);
+  };
+
+  const debounceCallback = useCallback(debounce(DEBOUNCE_DELAY, onQueryChange), []);
```
Comment (pernielsentikaer) [inline extensions/kube-docs/src/index.tsx:29]:
```suggestion
```
You shouldn't use a “custom” debounce, there is one handling this directly in Raycast, you can add `throttle` directly to the [<List> element](https://developers.raycast.com/api-reference/user-interface/list#list)

File: extensions/kube-docs/src/index.tsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
+import { Action, ActionPanel, Icon, List } from "@raycast/api";
+import { useCallback, useState } from "react";
+import { debounce } from "throttle-debounce";
+import { DocsItem, DocsResponse } from "./types";
+import useFetcher from "./hooks/useFetcher";
+
+const DEBOUNCE_DELAY = 300;
```
Comment (pernielsentikaer) [inline extensions/kube-docs/src/index.tsx:7]:
```suggestion
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @rpvsilva 👋 

Thanks for your contribution 💪 

I have some feedback for you to address 💯 

Is there a way to add a list of “default” elements, so the list is not empty, like many other documentation extensions:

wQ9pVXSRyzzy7Izn

Just trying to see if we can find a way to spice the UX a little bit 😄 

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by rpvsilva** [comment]
> Hi @rpvsilva 👋
> 
> Thanks for the update, much better 🔥

Thank you for your review and your suggestions! I think the extension is more complete now

**Pull request merged**